engine/consensus/compliance/engine.go

```diff
@@ -179,8 +179,15 @@ func (e *Engine) Ready() <-chan struct{} {
 		e.unit.Launch(e.finalizationProcessingLoop)
 
 		ctx, cancel := context.WithCancel(context.Background())
-		signalerCtx, _ := irrecoverable.WithSignaler(ctx)
+		signalerCtx, hotstuffErrChan := irrecoverable.WithSignaler(ctx)
```

Comment:
Nice catch. 

I wonder how did you catch this? Is it because the empty collection somehow causing hotstuff to fail?
```suggestion
		signalerCtx, hotstuffFatalErrChan := irrecoverable.WithSignaler(ctx)
```

Reply:
Exactly - for the reason you pointed out before: https://github.com/onflow/flow-go/pull/2523#discussion_r887161191

Reply:
Oh I see.